spec/scenarios/billing/billing_monthly_spec.rb

```diff
@@ -0,0 +1,296 @@
+      expect(last_invoice_inv_sub.timestamp).to match_datetime("2024-03-01T00:00:00Z")
+
+      # travel to the middle of month and create events per each charge:
+      events_date = DateTime.new(2024, 3, 15)
+      travel_to(events_date) do
+        [billable_metric_metered, billable_metric_recurring].each do |billable_metric|
+          create_event(
+            {
+              transaction_id: SecureRandom.uuid,
+              external_subscription_id: subscription.external_id,
+              code: billable_metric.code,
+              properties: {
+                billable_metric.field_name => 10
+              }
+            }
+          )
+        end
+        perform_all_enqueued_jobs
+      end
+      # we should create invoices for received pay_in_advance charges:
+      expect(subscription.reload.invoices.count).to eq(4)
+      last_invoices = subscription.invoices.order(:created_at).last(2)
+      expect(last_invoices.map { |inv| inv.fees.charge.count }.uniq).to eq([1])
+      expect(last_invoices.map { |inv| inv.fees.charge.map(&:charge_id) }).to match_array([[charge_metered_not_prorated_in_advance.id], [charge_recurring_prorated_in_advance.id]])
+      expect(last_invoices.map { |inv| inv.total_amount_cents }.sort).to match_array([100000, 50000000])
+
+      billing_time = DateTime.new(2024, 4, 1)
+      travel_to(billing_time) do
+        EventsRecord.connection.commit_db_transaction
+        perform_billing
+      end
+      expect(subscription.reload.invoices.count).to eq(5)
+      last_invoice = subscription.invoices.order(:created_at).last
+      # 0 is only for now!
+      expect(last_invoice.fees.fixed_charge.count).to eq(0)
+      # note that charge_recurring_prorated_in_advance should be included, because since it's recurring, it has usage,
+      # ewhich we're chargin in_advance
+      expect(last_invoice.fees.charge.count).to eq(3)
+      expect(last_invoice.fees.charge.map(&:charge_id)).to match_array([charge_metered_not_prorated_in_arrears.id, charge_recurring_prorated_in_arrears.id, charge_recurring_prorated_in_advance.id])
+
+      # check amounts by charges
+      metered_not_prorated_fee = last_invoice.fees.charge.find { |fee| fee.charge_id == charge_metered_not_prorated_in_arrears.id }
+      recurring_prorated_fee = last_invoice.fees.charge.find { |fee| fee.charge_id == charge_recurring_prorated_in_arrears.id }
+      recurring_prorated_advance_fee = last_invoice.fees.charge.find { |fee| fee.charge_id == charge_recurring_prorated_in_advance.id }
+      expect(metered_not_prorated_fee.amount_cents).to eq(10_000)
+      # this should be 500000 * 10 * 17/31,
+      # prorated_fee_amount = 2_741_935 - this is math correct, but service returns 2_741_940 because of rounding (10 * 17 / 31)...
+      prorated_fee_amount = 2_741_940
+      expect(recurring_prorated_fee.amount_cents).to eq(prorated_fee_amount)
+      expect(recurring_prorated_advance_fee.amount_cents).to eq(50_000_000)
```

Comment:
same here, simpler and easier to read and understand that is going on here...

```suggestion
      # this should be 500000 * 10 * 17/31,
      # prorated_fee_amount = 2_741_935 - this is math correct, but service returns 2_741_940 because of rounding (10 * 17 / 31)...
      prorated_fee_amount = 2_741_940

      expected_charge_fees = [
        { charge_id: charge_metered_not_prorated_in_arrears.id, amount_cents: 10_000 },
        { charge_id: charge_recurring_prorated_in_arrears.id, amount_cents: prorated_fee_amount },
        { charge_id: charge_recurring_prorated_in_advance.id, amount_cents: 50_000_000 }
      ]

      actual_charge_fees = last_invoice.fees.charge.map do |fee|
        {
          charge_id: fee.charge_id,
          amount_cents: fee.amount_cents
        }
      end

      expect(actual_charge_fees).to match_array(expected_charge_fees)
```